packages/edit-post/src/components/sidebar/post-initial-panel/index.js

```diff
@@ -0,0 +1,126 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	PanelBody,
+	__experimentalText as Text,
+	__experimentalVStack as VStack,
+} from '@wordpress/components';
+import {
+	store as editorStore,
+	privateApis as editorPrivateApis,
+} from '@wordpress/editor';
+import { decodeEntities } from '@wordpress/html-entities';
+import { useSelect } from '@wordpress/data';
+import { useMemo } from '@wordpress/element';
+import { page as pageIcon } from '@wordpress/icons';
+import { __, sprintf } from '@wordpress/i18n';
+import { humanTimeDiff } from '@wordpress/date';
+import { addQueryArgs } from '@wordpress/url';
+
+/**
+ * Internal dependencies
+ */
+import { unlock } from '../../../lock-unlock';
+
+const {
+	PostActions: EditorPostActions,
+	postManagementActions,
+	PostSidebarCard,
+} = unlock( editorPrivateApis );
+
+const {
+	trashPostAction,
+	usePermanentlyDeletePostAction,
+	useRestorePostAction,
+	postRevisionsAction,
+} = postManagementActions;
+
+function PostActions( { postType, postId } ) {
```

Comment:
What is specific about edit-post that is forcing us to override the default "editor" PostActions?

Reply:
When we execute an action, it may require us to navigate to a different page. For example, when we delete a post while editing it, we need to navigate to a completely different page using "window.location.href". This is a unique requirement on the edit-post. We are not replacing the action, but rather, we are adding specific callbacks to the onPerform function when needed.

Reply:
But it's almost always the same kind of override. So rather than all of this complex code, some simple `onNavigateToList`/`onNavigateToPost` props or something like that could work.

Reply:
> But it's almost always the same kind of override. So rather than all of this complex code, some simple `onNavigateToList`/`onNavigateToPost` props or something like that could work.

That would work. Thank you for the suggestion @youknowriad 👍 Another possibility is a a generic usePostActions( onPerform ) with a global onPerform( actionId, items) that receives the actionId and the items where the action was executed. And then the usage would be very simple:
```
usePostActions( onPerform( actionId, items)=> {
switch(actionId)
case: moveToTrass
case: permanentlyDelete:
onNavigateToList();
break:
case: editPost:
navigateToPost
} );
```

Let me know which API would you prefer a onNavigateToList and a onNavigateToPost or a global onPerform.

Reply:
Maybe the generic `onPerform` is slightly better.

Reply:
* `onPerform` is not very obvious. Maybe `onNavigateAfterAction`? `onAfterAction`? `onActionPerformed`?
* I wonder whether that API is too powerful, though. Concerns:
    * For the reader, how obvious is it that this prop is meant to strictly set whether and where to route after the action is completed?
    * Is there a clearer and stricter alternative?
    * Does this introduce indirection, in the sense that the reader needs to juggle two distinct components to understand the whole cascade of effects?
    * (This is not a strong opinion, just caution.)

Reply:
Good points. The onPerform may be used for other things besides navigation e.g: close a menu after the action is performed etc. And we already had onPerform for actions before:
https://github.com/WordPress/gutenberg/blob/9c0a5ac3e76fc6298d6ad13fa389d5ee834d77ba/packages/edit-site/src/components/actions/index.js#L150-L153

But not all of the actions had it.

I think I will go for onActionPerformed, it more clear on what it does and what is expected when compared with just onPerform.